ext/drmgr/drmgr.c

```diff
@@ -704,8 +705,11 @@ drmgr_bb_event(void *drcontext, void *tag, instrlist_t *bb, bool for_trace,
 
     /* Pass 3: instru, per instr */
     pt->cur_phase = DRMGR_PHASE_INSERTION;
-    pt->first_app = instrlist_first(bb);
-    pt->last_app = instrlist_last(bb);
+    pt->first_instr = instrlist_first(bb);
+    pt->first_nonlabel_instr = instrlist_first(bb);
+    while (instr_is_label(pt->first_nonlabel_instr))
```

Comment:
There are two "find the first nonlabeled instruction" loops; do you think it's worth adding it as a central helper, e.g. an `instrlist_first_nonlabel` ?

Reply:
Sure, SGTM.  Added.  PTAL.